### ISSUE TITLE
Code equivalency: "it" variable overwritten in closures without parameters.

### ISSUE THREAD

**Issue by apryamostanov** [INFINITE-TECHNOLOGY/BLACKBOX#1]
The following code fails in current snapshot:

```
package io.infinite.blackbox.issues

import io.infinite.blackbox.BlackBox
import io.infinite.blackbox.BlackBoxLevel

class Issue001 {

    @BlackBox(blackBoxLevel = BlackBoxLevel.EXPRESSION)
    void test() {
        Object[] objects = [1,2,3,4,5]
        objects.each {
            assert it != null
        }
    }

}
```

**Comment by apryamostanov** [comment]
Issue happens because of [Standard Expression Transformation](https://github.com/INFINITE-TECHNOLOGY/BLACKBOX/wiki#standard-expression-transformation):

Expression is transformed to ClosureExpression and passed to BlackBoxEngine.expressionEvaluation method.
However issue happens in case of ClosureExpression without declared parameters.
Wrapping ClosureExpression with another ClosureExpression overrides the implicit "it" variable.

To fix this it is just needed to prevent redeclaration of "it" parameter in automatically generated closure wrappers - by declaring some placeholder parameter (with unique name per closure to avoid duplicates).

**Comment by apryamostanov** [comment]
Fixed in 82732bef3111849cf618ef7f982d09edef8997e1
